site/global-guides/voice/authentication.mdx

```diff
@@ -0,0 +1,75 @@
+---
+id: inAppAuthGuide
+title: Authentication for Global In-App Calling
+slug: /voice/guides/inAppCalling/authentication
+description: Authentication In-App Calling (WebRTC)
+keywords:
+  - bandwidth
+  - voice
+hide_title: false
+image: '@site/static/img/bw-icon.svg'
+---
+## Retrieving an access token
+
+:::caution
+
+The token should be retrieved from a server running in a secure environment and securely provided to clients. Client-side javascript does not have a mechanism for hiding these credentials so DO NOT place these directly in your client-side code.
+
+Bandwidth accepts no responsibility for the loss of account credentials and any resulting network traffic, fraud, or undesired account access that results from failing to manage account access credentials in a completely secure manner.
+:::
+
+
+The process to retrieve an access token you require you to have your ClientID and ClientSecret which you generated in the UI. You will then make a call to our token endpoint with these credentials. We will then return to you an `access_token`, `id_token`, `token_type`, and `expires_in` (this is the number of seconds before the token will expire).
+
+```mermaid
+sequenceDiagram
+    Application->>BW Global API AuthServer: GET /api/v1/oauth2/token?scope=voice.webrtc <br> Headers: { Authorization: Basic PEJhc2ljIEF1dGggVXNlcm5hbWU+OjxCYXNpYyBBdXRoIFBhc3N3b3JkPg== }
+    BW Global API AuthServer->>BW Global API AuthServer: Validate credentials
+    BW Global API AuthServer->>Application: 200 OK <br> {"access_token": "...", "access_token_id": "...", "token_type": "Bearer", "expires_in": 600}
+    Application->>WebRTC Gateway: GET /someresource <br> Headers: {Authorization: <token_type> <access_token>}
+```
+### Request
+In order to retrieve the token you will use Basic Auth `ClientID:ClientSecret`. Below is a sample request
+
+```sh
+curl --request POST \
+  --url https://<voxbones_token_providerd_url>/api/v1/oauth2/token?scope=voice.webrtc \
+  --header 'Authorization: Basic PEJhc2ljIEF1dGggVXNlcm5hbWU+OjxCYXNpYyBBdXRoIFBhc3N3b3JkPg==' \
+  --header 'Content-Type: application/x-www-form-urlencoded' \
+  --data grant_type=client_credentials
+```
+Now that we've shown the request let's break down the parts of the call.
+
+- **url** `https://<voxbones_token_provider_url>/api/v1/oauth2/token?scope=voice.webrtc` - This is the BW Global API AuthProvider's `token_endpoint`. You may notice there is a query parameter of `scope` this is used to limit the permissions in the token. When you request a token with the scope parameter that token will **ONLY** be able to talk to resources that require those particular scopes which were requested. In our example, we are limiting the token to only being able to talk to the WebRTC Gateway. If you do not provide a scope then the token will be minted for ALL the scopes that the credentials are allowed to have as defined during creation.
```

Comment:
```suggestion
- **url** `https://<voxbones_token_provider_url>/api/v1/oauth2/token?scope=voice.webrtc` - This is the BW Global API AuthProvider's `token_endpoint`. **IMPORTANT:** There is a query parameter of `scope`. This is used to limit the permissions in the token. When you request a token with the scope parameter, that token will **ONLY** be able to access resources that require the specified scopes. In our example, we are limiting the token's permissions to only the WebRTC Gateway. If you do not provide a scope, then the token will be minted for ALL scopes that are allowed by the credentials when they were created.
```

Reply:
- The `<voxbone url>` will need replacing.
- You used the term `AuthProvider` but you were saying `AuthServer` prior to here. If you agree with my previous terminology of `In-App Calling Auth Server`, then it won't matter.